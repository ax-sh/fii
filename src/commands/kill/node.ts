import type { GluegunCommand } from 'gluegun'

import type { ExtendedToolbox } from '../../types'

const command: GluegunCommand<ExtendedToolbox> = {
  name: 'node',
  alias: ['bun'],
  description: 'Kill all node process',
  run: async (toolbox) => {
    const { print } = toolbox
    const spinner = print.spin('killing all node processes')

    try {
      await toolbox.killProcess('node.exe')
      await toolbox.killProcess('bun.exe')
    } catch {
      return spinner.warn('Killing failed')
    }

    spinner.succeed('Done')
  },
}

module.exports = command
