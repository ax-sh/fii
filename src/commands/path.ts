import type { GluegunCommand } from 'gluegun'

import { type ExtendedToolbox } from '../types'

const command: GluegunCommand<ExtendedToolbox> = {
  name: 'path',
  run: async (toolbox) => {
    const { print, filesystem } = toolbox
    const cliProjectPath = filesystem.path(__dirname, '..', '..', '..')
    const spinner = print.spin(`Location`)

    spinner.succeed(`Fii location: ${cliProjectPath}`)
  },
}

module.exports = command
