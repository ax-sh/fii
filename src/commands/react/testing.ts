import type { GluegunCommand } from 'gluegun'

import { type ExtendedToolbox } from '../../types'

const command: GluegunCommand<ExtendedToolbox> = {
  name: 'testing',
  description: 'Adds react testing lib with vitest',
  run: async (toolbox) => {
    const { print } = toolbox

    const spinner = print.spin(`Adding react testing lib`)
    const rt = await import('../../lib/helpers/react-utils')
    // await reactTesting.addDeps()
    await rt.addRTLToVitest()

    // await reactTesting.setupTsTypes()

    spinner.succeed(`Added react testing lib testing`)
  },
}

module.exports = command
