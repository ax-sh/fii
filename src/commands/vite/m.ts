import type { GluegunCommand } from 'gluegun'

const command: GluegunCommand = {
  name: 'm',
  description: 'Create a bare minimum vite react ts project',
  run: async (toolbox) => {
    const { print, parameters, system, filesystem } = toolbox
    const appName = parameters.first

    const installedPath = filesystem.path(appName)
    if (filesystem.isDirectory(installedPath)) {
      return print.error(`Folder with the name [${appName}] already exists`)
    }

    const spinner = print.spin(`Creating vite app with react swc and tailwind [${appName}]`)
    await system.run(`bun create vite ${appName} --template react-swc-ts`)
    spinner.info(`Created vite app [${appName}]!`)
    process.chdir(installedPath)

    spinner.info(`Installing clsx and prettier!`)
    await system.run('bun i && ni clsx && fii add prettier && fii add tailwind && fii add release')

    const { gitFlowInit } = await import('../../lib/helpers/git-utils')
    const gitInit = await gitFlowInit()
    spinner.succeed(gitInit)

    spinner.succeed(`Created App ${appName}`)
    print.info(`webstorm ${installedPath}`)
    await system.run(`webstorm ${installedPath}`)
  },
}

module.exports = command
