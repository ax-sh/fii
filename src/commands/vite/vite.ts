import type { GluegunCommand } from 'gluegun'

const command: GluegunCommand = {
  name: 'vite',
  run: async (toolbox) => {
    const { print, parameters, system } = toolbox
    const name = parameters.first

    print.info(`Todo vite ${name}`)
    await system.run('echo ni -D husky')
  },
}

module.exports = command
