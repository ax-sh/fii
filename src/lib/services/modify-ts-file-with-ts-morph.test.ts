import * as path from 'node:path'
import { Project, ScriptKind, SourceFile, SyntaxKind } from 'ts-morph'

import {
  addImports,
  addImportsToSourceFile,
  addVitePlugins,
  formatSourceFileToString,
  getDefineConfigFromSourceFile,
} from './parse-ts'

describe('Import Management', () => {
  let project: Project
  let sourceFile: SourceFile

  beforeEach(() => {
    project = new Project({ useInMemoryFileSystem: true })
  })

  const createSource = (code: string) => {
    const resolvedPath = 'test.ts'
    sourceFile = project.createSourceFile(resolvedPath, code, {
      scriptKind: path.extname(resolvedPath) === '.ts' ? ScriptKind.TS : ScriptKind.JS,
    })
    return sourceFile
  }

  test('should add new imports to empty file without duplicates', () => {
    const sf = createSource('const x:number = 3')

    addImportsToSourceFile(sf, [
      { from: 'react', imports: 'React' },
      { from: 'react', imports: 'React' },
      { from: 'react', imports: ['useState'] },
      { from: 'react', imports: ['useState', 'useEffect'] },
      { from: 'react', imports: ['useEffect'] },
    ])
    const raw = formatSourceFileToString(sf)
    expect(raw).toMatchSnapshot()
    const imports = sf.getImportDeclarations()

    expect(imports).toHaveLength(1)
    expect(imports[0].getModuleSpecifierValue()).toBe('react')
    expect(imports[0].getDefaultImport()?.getText()).toBe('React')
  })

  test('should add multiple imports types', () => {
    const sf = createSource(`export const x = 10;`)

    addImports(sf, [
      { moduleSpecifier: 'react', defaultImport: 'React' },
      { moduleSpecifier: 'components', namedImports: ['Button', 'Input'] },
      { moduleSpecifier: 'lib', defaultImport: 'lib', namedImports: ['util'] },
    ])
    const imports = sf.getImportDeclarations()
    expect(imports).toHaveLength(3)
    expect(imports[0].getDefaultImport().getText()).toEqual('React')
    expect(imports[1].getNamedImports()[0].getName()).toEqual('Button')
    expect(imports[2].getNamedImports()[0].getName()).toEqual('util')
    const output = sf.getFullText()
    expect(output).toMatchSnapshot()
  })

  test('should not duplicate existing imports', () => {
    const sf = createSource(`
      import React from 'react';
      export const x = 10;
    `)

    addImportsToSourceFile(sf, [
      { from: 'react', imports: 'React' },
      { from: 'react', imports: ['useState'] },
    ])
    console.log(formatSourceFileToString(sf))

    const imports = sf.getImportDeclarations()
    expect(imports).toHaveLength(1)
    expect(imports[0].getNamedImports()[0].getName()).toBe('useState')
  })

  test('should check if modify plugins property exists', () => {
    const sf = createSource(`
import react from '@vitejs/plugin-react';
import UnoCSS from 'unocss/vite';
import { defineConfig } from 'vite';
// https://vite.dev/config/
export default defineConfig({

});`)
    const configObject = getDefineConfigFromSourceFile(sf)
    const newBasePath = 'foo'
    // Check if the `base` property exists
    const baseProperty = configObject.getProperty('base')
    if (!baseProperty || !baseProperty.isKind(SyntaxKind.PropertyAssignment)) {
      // If the `base` property doesn't exist, create it with a default value
      configObject.addPropertyAssignment({
        name: 'base',
        initializer: `"${newBasePath}"`, // Default value for `base`
      })
      console.log("`base` property was added with a default value of '/'")
    } else {
      // If the `base` property doesn't exist, create it with the new value
      configObject.addPropertyAssignment({
        name: 'base',
        initializer: `"${newBasePath}"`, // Default value for `base`
      })
      console.log('`base` property was added with a default value:', newBasePath)
    }
    const updatedContent = formatSourceFileToString(configObject.getSourceFile())
    expect(updatedContent).toContain(`base: "${newBasePath}"`)
    console.log(updatedContent)
  })

  const viteConfig = `
import react from '@vitejs/plugin-react';
import UnoCSS from 'unocss/vite';
import { defineConfig } from 'vite';
// https://vite.dev/config/
export default defineConfig({
plugins: [UnoCSS(), react()],
});`

  test('should modify plugins', () => {
    const sf = createSource(viteConfig)
    const modified = getDefineConfigFromSourceFile(sf)

    console.log(formatSourceFileToString(modified.getSourceFile()))
  })

  test('should modify plugins without duplicates', () => {
    const sf = createSource(viteConfig)
    const modified = addVitePlugins(sf, ['dff', 'daff'])

    console.log(formatSourceFileToString(modified.getSourceFile()))
  })
})
