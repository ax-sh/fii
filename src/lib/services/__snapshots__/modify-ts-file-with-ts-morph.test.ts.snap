// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Import Management > should add multiple imports types 1`] = `
"import React from "react";
import { Button, Input } from "components";
import lib, { util } from "lib";

export const x = 10;"
`;

exports[`Import Management > should add new imports to empty file 1`] = `
"import React, { useState, useEffect } from "react";

const x: number = 3;
"
`;

exports[`Import Management > should add new imports to empty file 2`] = `
"import React from "react";

const test = 3;
"
`;

exports[`Import Management > should add new imports to empty file without duplicates 1`] = `
"import React, { useState, useEffect } from "react";

const x: number = 3;
"
`;
